core/src/main/java/io/questdb/griffin/model/QueryModel.java

```diff
@@ -565,6 +571,10 @@ public void copyDeclsFrom(LowerCaseCharSequenceObjHashMap<ExpressionNode> decls)
         }
     }
 
+    public void copyHints(LowerCaseCharSequenceObjHashMap<CharSequence> hints) {
+        this.hintsMap.putAll(hints);
```

Comment:
i would stick asssert here that hintsMap is empty to begin with, so this method is harder to abuse

Reply:
the target model does not have to be empty. consider unions. consider this (one of the queries in tests): 
```
select /*+ HINT_A*/ a,b,c 
from xyz 
where a = 1 
union 
select /*+ HINT_B*/ a,b,c 
from xyz
where a = 2
``` 

we push `HINT_A` to the union model even the union model has its own hint. So the union model will have both `HINT_A` and `HINT_B`. (while the main model has just `HINT_A`)